### DISCUSSION TITLE
Multi-user registration

### DISCUSSION THREAD

**Discussion by Flxshin** [navidrome/navidrome#3512] [Ideas]
It is troublesome to create users manually every time. Is it possible to directly provide open registration?

**Comment by ihatenodejs** [comment]
I'm taking a look at the API as I would like to implement this as well.

**Reply by battlemag3** [reply]
how is it going?

**Comment by olivertzeng** [comment]
any updates?

**Reply by metalheim** [reply]
What would even be the use-case? Navidromes focus is beeing a personal music streaming application.
You shouldn't have so many users that it becomes troublesome to create them.

I mean im not telling anyone how to use navidrome, but don't expect development in directions that aren't focus of the project.
If you really need to handle a lot of users for your use-case, then you can use an External Authentication Provider:
https://www.navidrome.org/docs/getting-started/extauth-quickstart/

To quote the navidrome documentation:
```
This approach is usually called “reverse proxy authentication”, and offers several benefits:

Increased flexibility: Leverage authentication methods not supported by Navidrome (LDAP, OAuth, 2FA, etc.)
Single Sign-On (SSO): Users can login once and access multiple services
Centralized user management: Manage all your users in one place
```

**Reply by olivertzeng** [reply]
> What would even be the use-case? Navidromes focus is beeing a personal music streaming application. You shouldn't have so many users that it becomes troublesome to create them.
> 
> I mean im not telling anyone how to use navidrome, but don't expect development in directions that aren't focus of the project. If you really need to handle a lot of users for your use-case, then you can use an External Authentication Provider: [navidrome.org/docs/getting-started/extauth-quickstart](https://www.navidrome.org/docs/getting-started/extauth-quickstart/)
> 
> To quote the navidrome documentation:
> 
> ```
> This approach is usually called “reverse proxy authentication”, and offers several benefits:
> 
> Increased flexibility: Leverage authentication methods not supported by Navidrome (LDAP, OAuth, 2FA, etc.)
> Single Sign-On (SSO): Users can login once and access multiple services
> Centralized user management: Manage all your users in one place
> ```

I have a bunch of friends that i want to share my service to, it would be so much easer if this is implemented
